docs/books/user-guide/routing.adoc

```diff
@@ -102,6 +102,17 @@ Link routing supports local transactions to a broker.
 +
 With a link route, consumers can provide server-side selectors for broker subscriptions.
 
+[id='considerations-for-broker-connection-failure-handling']
+=== Considerations for Broker Connection Failure Handling
+
+The message routing and link routing mechanisms both enable you to transfer messages to brokers. However, they differ in how they behave if the connection to the broker is lost. If you require the ability for the connection to be reestablished automatically, you should use message routing (with autolinks).
+
+With link routing::
+The client appears to be directly connected to the broker even though it is actually connected to a router, which is link routed to the broker. If the broker goes offline, the router sees the connection failure and closes all of the routed links that were attached to the broker. The client's connection to the router, however, remains open. Therefore, the client does not see the connection failure between the broker and the router and does not attempt to reestablish it.
+
+With message routing using autolinks::
+The client is connected to a router, and the router is connected to the broker. Therefore, if the broker goes offline, the router will see the connection failure and attempt to reestablish the connection when the broker comes back online.
```

Comment:
In the case of message routing, the router network will automatically reroute deliveries to alternate destinations if they are available.  There will likely be some deliveries that are returned to the sender with RELEASED disposition meaning that they were not delivered anywhere.  Deliveries may also come back with MODIFIED/undeliverable disposition indicating that they _may_ have been delivered, but were not settled.

If there is only one broker in a message-routing/autolink deployment and that broker goes offline, senders to the broker will see their deliveries RELEASED and eventually will experience backpressure (no new credit) until the broker comes back online.